### PR TITLE
Redefine task and its context

### PR DESCRIPTION
**Pull request by hakonhall**
After discussions with bratseth, we'll not organize tasks in a tree structure,
and instead have tasks mostly 1:1 with components. Tasks are put in chains, and
may have ordering constraints between them that the chain resolves.

The FileSystem API has been removed and instead use the raw java NIO
FileSystem, and use Jimfs to mock FileSystem in tests.

NodeAdminStateUpdater has been made an interface, since the HTTP request
handler needs to delegate to it when getting requests. Much of the public HTTP
API will get internalized with standalone NodeAdmin.

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/provider/NodeAdminStateUpdater.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.provider;// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/provider/NodeAdminStateUpdater.java]: Several files have a duplicate header.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/TaskComponent.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    protected TaskComponent(ComponentId id) {
+        super(id);
+    }
+
+    public String name() {
+        return getIdString();
+    }
+
+    /**
+     * Execute an administrative task to converge the system towards some ideal state.
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/TaskComponent.java]: Keep this javadoc on interface only?

File: jdisc_core/src/main/java/com/yahoo/jdisc/Timer.java [hunk 3]
```diff
@@ -27,4 +29,10 @@
      */
     long currentTimeMillis();
 
+    /**
+     * Convenience method for getting an java.util.Instance from currentTimeMillis().
+     */
+    default Instant currentTime() {
```
Comment (freva) [inline jdisc_core/src/main/java/com/yahoo/jdisc/Timer.java]: This isn't used anywhere?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/AdminComponent.java [hunk 4]
```diff
@@ -13,6 +15,11 @@
      */
     void enable();
 
+    /**
+     * @return NodeAdminStateUpdater used by the REST API
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/AdminComponent.java]: Why is this needed? NodeAdminStateUpdater is injected into REST API becaue NodeAdminProvider is Provider of NodeAdminStateUpdater?

**Comment by hakonhall** [comment]
the reason I got those failures was because the merge was stopped at the first conflict - up to my commits, but didn't apply the last changes on HEAD - a revert. This has now been cherry-picked and hopefully the tests will pass.

**Comment by hakonhall** [comment]
@freva PTAL

**Review by freva — APPROVED** [review]
